### PR TITLE
Added License

### PR DESCRIPTION
**Pull request by xquek**
Added ability to add license using extras json file.

### CODE REVIEW

**Comment by xquek** [comment]
@orodeh 

This is ready for review. I had done the following

1. Add new case classes: `DxAttrs`, `DxDetails` `DxLicense`
2. Place DxRunSpec and  as member of DxAttrs

File: src/main/scala/dxWDL/compiler/GenerateIRWorkflow.scala [hunk 1]
```diff
@@ -412,13 +412,7 @@ case class GenerateIRWorkflow(wf : WorkflowDefinition,
...
         }
 
-        val applet = IR.Applet(s"${wfName}_frag_${genFragId()}",
-                               inputVars,
-                               outputVars,
-                               IR.InstanceTypeDefault,
-                               IR.DockerImageNone,
-                               IR.AppletKindWfFragment(innerCall.toVector, blockPath, fqnDictTypes),
-                               wfSourceStandAlone)
+        val applet = IR.Applet(s"${wfName}_frag_${genFragId()}", inputVars, outputVars, IR.InstanceTypeDefault, IR.DockerImageNone, IR.AppletKindWfFragment(innerCall.toVector, blockPath, fqnDictTypes), wfSourceStandAlone)
```
Comment (orodeh) [inline src/main/scala/dxWDL/compiler/GenerateIRWorkflow.scala]: long line, I prefer the older style.

File: src/main/scala/dxWDL/compiler/GenerateIRWorkflow.scala [hunk 2]
```diff
@@ -527,13 +521,8 @@ case class GenerateIRWorkflow(wf : WorkflowDefinition,
...
         val outputVars: Vector[CVar] = inputVars
-        val applet = IR.Applet(s"${wfName}_${COMMON}",
-                               inputVars,
-                               outputVars,
-                               IR.InstanceTypeDefault,
-                               IR.DockerImageNone,
-                               IR.AppletKindWfInputs,
-                               wfSourceStandAlone)
+
+        val applet = IR.Applet(s"${wfName}_${COMMON}", inputVars, outputVars, IR.InstanceTypeDefault, IR.DockerImageNone, IR.AppletKindWfInputs, wfSourceStandAlone)
```
Comment (orodeh) [inline src/main/scala/dxWDL/compiler/GenerateIRWorkflow.scala]: Long line.

File: src/main/scala/dxWDL/base/Extras.scala [hunk 3]
```diff
@@ -167,30 +168,88 @@ case class DxRunSpec(access: Option[DxAccess],
...
+    }
+}
+
+case class DxLicense(name: String,
+                     repoUrl: String,
+                     version: String,
+                     license: String,
+                     licenseUrl: String,
+                     author: String
+                    )
```
Comment (orodeh) [inline src/main/scala/dxWDL/base/Extras.scala]: This bracket would be better one line above.

File: src/main/scala/dxWDL/base/Extras.scala [hunk 4]
```diff
@@ -505,6 +591,8 @@ object Extras {
                perTaskDxAttrs,
                parseDockerRegistry(
                    checkedParseObjectField(fields, "docker_registry"),
-                   verbose))
+                   verbose),
+
```
Comment (orodeh) [inline src/main/scala/dxWDL/base/Extras.scala]: Unnecessary empty lines.

File: src/main/scala/dxWDL/compiler/GenerateIRWorkflow.scala [hunk 5]
```diff
@@ -574,13 +563,7 @@ case class GenerateIRWorkflow(wf : WorkflowDefinition,
...
         }.toVector
 
-        val applet = IR.Applet(s"${wfName}_${OUTPUT_SECTION}",
-                               inputVars.map(_.cVar),
-                               outputVars,
-                               IR.InstanceTypeDefault,
-                               IR.DockerImageNone,
-                               IR.AppletKindWfOutputs,
-                               wfSourceStandAlone)
+        val applet = IR.Applet(s"${wfName}_${OUTPUT_SECTION}", inputVars.map(_.cVar), outputVars, IR.InstanceTypeDefault, IR.DockerImageNone, IR.AppletKindWfOutputs, wfSourceStandAlone)
```
Comment (orodeh) [inline src/main/scala/dxWDL/compiler/GenerateIRWorkflow.scala]: Long line

**Comment by orodeh** [comment]
It looks very good, thanks! There are a few minor code style issues. Other than that, this is good to go. You have lines that go over 100 characters, I marked some of them. There some unnecessary empty lines, I marked some of those too.

**Comment by orodeh** [comment]
Great, I'll merge it.

**Pull request merged**